### PR TITLE
Improve Acunetix360 Parser

### PR DESCRIPTION
**Pull request by kiliczsh**
- Improve filename detection
- Improve CWE detection
- Improve states

### CODE REVIEW

**Comment by kiliczsh** [comment]
@valentijnscholten @damiencarol can you give me more details about integration tests (tests/dedupe_test.py)?

**Comment by valentijnscholten** [comment]
> @valentijnscholten @damiencarol can you give me more details about [failure ](https://github.com/DefectDojo/django-DefectDojo/runs/5360504664?check_suite_focus=true#step:8:2106) in integration tests (tests/dedupe_test.py)?

that test sometimes fails without clear reason, I have restarted the testsuite.

**Comment by Maffooch** [comment]
@kiliczsh please base your pr against the dev branch

**Comment by kiliczsh** [comment]
@Maffooch thanks for your warning 👍🏼

File: dojo/tools/acunetix360/parser.py [hunk 1]
```diff
@@ -17,14 +17,24 @@ def get_description_for_scan_types(self, scan_type):
         return "Acunetix360 JSON format."
 
     def get_findings(self, filename, test):
-        data = json.load(filename)
+        tree = filename.read()
+        try:
+            data = json.loads(str(tree, 'utf-8-sig'))
+        except:
+            data = json.loads(tree)
```
Comment (damiencarol) [inline dojo/tools/acunetix360/parser.py:24]: json.load() method support uTF8 / UTF16 wi or without bom.
This change break this support.

File: dojo/tools/acunetix360/parser.py [hunk 2]
```diff
@@ -51,6 +61,16 @@ def get_findings(self, filename, test):
...
 
+            if item["State"].find("FalsePositive") != -1:
+                finding.active = False
+                finding.verified = False
+                finding.false_p = True
+                finding.mitigated = None
+                finding.is_mitigated = False
+
+            if item["State"].find("AcceptedRisk") != -1:
+                finding.risk_accepted = True
```
Comment (damiencarol) [inline dojo/tools/acunetix360/parser.py:72]: I think that for this use case you should put `verified` at true and `active` to false

**Review by damiencarol — CHANGES_REQUESTED** [review]

File: dojo/tools/acunetix360/parser.py (hunk 1 above)
Comment (kiliczsh) [reply dojo/tools/acunetix360/parser.py]: rollback to previous version

File: dojo/tools/acunetix360/parser.py (hunk 2 above)
Comment (kiliczsh) [reply dojo/tools/acunetix360/parser.py:68]: updated verified and active fields too

**Review by damiencarol — CHANGES_REQUESTED** [review]
Modifications seems good. we will need data and new unit tests to cover the new code

**Comment by kiliczsh** [comment]
> Modifications seems good. we will need data and new unit tests to cover the new code

@damiencarol the scan files already have the [data](https://github.com/kiliczsh/django-DefectDojo/blob/acunetix360-maintenance/dojo/unittests/scans/acunetix360/acunetix360_one_finding.json#L34) and [unit tests](https://github.com/kiliczsh/django-DefectDojo/blob/acunetix360-maintenance/dojo/unittests/tools/test_acunetix360_parser.py#L49).

**Pull request closed without merging**